src/components/NetworkGraph.vue

```diff
@@ -120,8 +128,14 @@ export default {
   },
   methods: {
     nodeButtonClicked: function () {
+      this.$refs.overlay.style.display = 'block'
+      this.$refs.loader.style.display = 'block'
       const url = 'http://localhost:8000/api/explorer/' + this.metric
-      fetch(url).then(response => response.json()).then(data => this.readJson(data))
+      fetch(url).then(response => response.json()).then(data => {
+        this.$refs.overlay.style.display = 'none'
+        this.$refs.loader.style.display = 'none'
+        this.readJson(data)
```

Comment:
Shouldn't the parsing also be included in the time when the loading is shown?
Hence, change the visiblity only once the readJson finished.

Reply:
Also, `readJson` is a bad name as well. First of all, `read` implies an operation with no side-effects, hence, I'd expect to get something returned, but the value is either dropped or there is no return. 
Further, `Json` doesn't convey anything other than that we are talking about JSON data. 
Which brings us to the next issue: the `data` parameter and the useless unpacking coroutine. We could compress this thing into this without losing any information:

```Javascript
   fetch(url).then(response => {
        this.$refs.overlay.style.display = 'none'
        this.$refs.loader.style.display = 'none'
        this.read(response.json())
    }
```

We'd still have the bad `read`, but at least some noise is gone. Also, we'd wonder, what we are reading.